### ISSUE TITLE
[AWS Power Up] Quantum sea - Classifying water molecules and sodium ions in protein structures

### ISSUE THREAD

**Issue by shadow1229** [XanaduAI/QHack2022#59]
Team Name:
Lindwurm

Project Description:
Quantum sea - Classifying water molecules and sodium ions in protein structures

The goal of the project is building quantum machine learning-based classifiers which can classifies water molecules and sodium ions present in the crystallographic structure of protein obtained by X-ray crystallography, as a kind of toy program for predicting physicochemical properties related with the protein. X-ray crystallography is mainly used to obtain the structure of a protein with high resolution, by using diffraction of X-ray due to electrons in the protein. Due to the nature of the method, small molecules, atoms or ions with the same number of electrons are likely to produce similar peaks. For example, water molecule, one of the small molecules abaundant in protein crystal structures, have 10 electrons, is likely to be confused with sodium ions which has 10 electrons. However, since water molecules does not have net charge, while sodium ions having positive net charge, the structure of proteins that can hold water molecules and sodium ions are likely to be different. From this, water molecules and sodium ions in X-ray crystallography can be distinguished.

In this project, convolutional natural network-based water-sodium ion classifier with input as a voxelized 3D image of the structure of carbon, nitrogen, and oxygen atoms from proteins or other compounds (exclude water) in a cube which center is located at a location where sodium ion or water molecule exists and size of 16Å and grid spacing of 0.5Å. In the first layer of the classifier, quanvolution neural network was used for
convolution and pooling of 32x32x32 grid a into 16x16x16 grid. 

Source code:
https://github.com/shadow1229/Qhack_2022/tree/main/Quantum_sea

Resource Estimate:
Quantum simulator in AWS is expected to generation of the result from the first quanvolution layer. For generation of the training data, 655.36 billions of shots in quantum simulator is expected, where the quanvolution layer is defined as two 4-qubit circuits with four random rotations and two layers. 

(800 input structures x 100 rotations x 16 x 16 x 16 voxels x 2 circuits x  1000 quantum simulation shots per each circuit)

**Comment by isaacdevlugt** [comment]
Thank you for your Power Up submission! As a reminder, the final deadline for your project is February 25 at 17h00 EST. Submissions should be done here: https://github.com/XanaduAI/QHack/issues/new?assignees=&labels=&template=open_hackathon.md&title=%5BENTRY%5D+Your+Project+Title

This issue will be closed shortly.

Good luck!
